resources/js/Components/EditModal.vue

```diff
@@ -0,0 +1,308 @@
+    if (props.dataSetType === 'users' && key === 'last_login') return false
+    
+    return true
+  })
+  
+  return fields.map(field => ({
+    key: field,
+    label: field.charAt(0).toUpperCase() + field.slice(1).replace('_', ' '),
+    type: getFieldType(field),
+    options: getFieldOptions(field),
+    validation: getFieldValidation(field),
+  }))
+})
+
+function getFieldType(field) {
+  switch (field) {
+  case 'status':
+  case 'type':
+  case 'role':
+  case 'shelter':
+    return 'select'
+  case 'age':
+  case 'capacity':
+  case 'current_occupancy':
+    return 'number'
+  case 'rating':
+    return 'number'
+  case 'email':
+  case 'user_email':
+    return 'email'
+  case 'ip_address':
+    return 'text'
+  case 'timestamp':
+    return 'datetime-local'
+  default:
+    return 'text'
+  }
+}
+
+function getFieldOptions(field) {
+  switch (field) {
+  case 'status':
+    if (props.dataSetType === 'pets') {
+      return ['Available', 'Adopted']
+    } else if (props.dataSetType === 'users') {
+      return ['Active', 'Inactive']
+    } else if (props.dataSetType === 'logs') {
+      return ['Success', 'Failed', 'Error']
+    }
+    return ['Available', 'Adopted', 'Pending', 'Active', 'Inactive', 'Success', 'Failed', 'Error']
+    
+  case 'type':
+    if (props.dataSetType === 'pets') {
+      return ['Dog', 'Cat', 'Other']
+    }
+    return ['Dog', 'Cat', 'Bird', 'Fish', 'Other']
+    
+  case 'role':
+    if (props.dataSetType === 'users') {
+      return ['Admin', 'Shelter Caretaker', 'User']
+    }
+    return ['Admin', 'User', 'Moderator']
+    
+  case 'shelter':
+    if (props.dataSetType === 'pets') {
+      return ['Happy Paws', 'Cat Care', 'Animal Haven', 'Pet Paradise', 'Furry Friends', 'Safe Haven', 'Companion Care', 'Animal Rescue']
+    }
+    return []
+    
+  default:
+    return []
+  }
+}
+
+function getFieldValidation(field) {
+  switch (field) {
+  case 'email':
+    return {
+      required: true,
+      pattern: /^[^\s@]+@[^\s@]+\.[^\s@]+$/,
+      message: 'Please enter a valid email address',
+    }
+  case 'ip_address':
+    return {
+      required: true,
+      pattern: /^(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/,
+      message: 'Please enter a valid IP address',
+    }
+  case 'user_email':
+    return {
+      required: true,
+      pattern: /^[^\s@]+@[^\s@]+\.[^\s@]+$/,
+      message: 'Please enter a valid email address',
+    }
+  case 'age':
+  case 'capacity':
+  case 'current_occupancy':
+    return {
+      required: true,
+      min: 0,
+      message: 'Value must be 0 or greater',
+    }
+  case 'rating':
+    return {
+      required: true,
+      min: 0,
+      max: 5,
+      message: 'Rating must be between 0 and 5',
+    }
+  case 'name':
+    return {
+      required: true,
+      minLength: 2,
+      message: 'Name must be at least 2 characters long',
+    }
+  default:
+    return {
+      required: false,
+    }
+  }
+}
```

Comment:
Maybe go with something like this?
```suggestion
const validationRules = {
  email: {
    required: true,
    pattern: /^[^\s@]+@[^\s@]+\.[^\s@]+$/,
    message: 'Please enter a valid email address',
  },
  user_email: {
    required: true,
    pattern: /^[^\s@]+@[^\s@]+\.[^\s@]+$/,
    message: 'Please enter a valid email address',
  },
  ip_address: {
    required: true,
    pattern: /^(?:...regex omitted...)$/,
    message: 'Please enter a valid IP address',
  },
  age: {
    required: true,
    min: 0,
    message: 'Value must be 0 or greater',
  },
  capacity: {
    required: true,
    min: 0,
    message: 'Value must be 0 or greater',
  },
  rating: {
    required: true,
    min: 0,
    max: 5,
    message: 'Rating must be between 0 and 5',
  },
  name: {
    required: true,
    minLength: 2,
    message: 'Name must be at least 2 characters long',
  },
}
  ```
  
  and then simplify
  `function getFieldValidation(field) {
  return validationRules[field] || { required: false }
}`